### PR TITLE
[CE-610] - Add api methodology on the methodologies taxonomies repo

### PR DESCRIPTION
**Pull request by DevAVIJIT7**
### User story

- As a TPM, I am able to select API as a methodology for Classic Pen Test. 
- As a researcher, the API methodology is shown to me if the engagement type is an API
- As a customer, I can see the API methodology when the engagement type is an API
methodology document: https://docs.google.com/document/d/1si7s9WS5ZSzMkEvv8FAVv9oIRYyf-cN_D-i8u1PdxXU/edit

- Jira: CE-610

### To Test:
Content must reflect what we have in the methodology document.

### CODE REVIEW

File: methodologies/api_testing.json [hunk 1]
```diff
@@ -0,0 +1,888 @@
...
+    "steps": [
+        {
+        "key": "information",
+        "title": "Information gathering",
+        "description": "",
+        "type": "checklist",
+        "items": [
+          {
+            "key": "search_engine_discovery_and_reconnaissance",
+            "title": "Conduct Search Engine Discovery and Reconnaissance for Information Leakage",
```
Comment (fedetaglia) [inline methodologies/api_testing.json:18]: I can't find this step on the doc and I can'f find here the first 4 items on that doc. am I missing something?

Comment (DevAVIJIT7) [reply methodologies/api_testing.json:18]: Yes, Thanks. I expect the first 4 items will not have `caption` and `vrt_category` as not provided in google doc.

Comment (fedetaglia) [reply methodologies/api_testing.json:18]: yep that sounds right

File: methodologies/api_testing.json [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+        "key": "information",
+        "title": "Information gathering",
+        "description": "",
+        "type": "checklist",
+        "items": [
+          {
+            "key": "review_documentation",
+            "title": "Review Documentation",
+            "description": "Review provided documentation or Swagger Files for any leaked or hidden endpoints.",
+            "caption": ""
```
Comment (DevAVIJIT7) [inline methodologies/api_testing.json:20]: has validation in `schema.json` file.

File: methodologies/api_testing.json [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+            "description": "Identify from hidden fields, parameters, methods HTTP header analysis",
+            "tools": "Burp proxy, ZAP, Tamper data"
+          },
+          {
+            "key": "execution_paths",
+            "title": "Map execution paths through application",
+            "caption": "OTG-INFO-007, WAHHM - Recon and Analysis",
+            "description": "Map the target application and understand the principal workflows.",
+            "tools": "Burp proxy, ZAP"
+          },
```
Comment (fedetaglia) [inline methodologies/api_testing.json:78]: where did you get this from?

File: methodologies/api_testing.json [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+            "key": "imap_smtp_injection",
+            "title": "IMAP/SMTP Injection",
+            "caption": "OTG-INPVAL-011, WAHHM - Test Handling of Input",
+            "description": "Identifying vulnerable parameters with special characters (i.e.: \\, ‘, “, @, #, !, |).\nUnderstanding the data flow and deployment structure of the client\nIMAP/SMTP command injection (Header, Body, Footer)",
+            "tools": "Burp Proxy, ZAP",
+            "vrt_category": "server_side_injection"
+          },
+          {
+            "key": "code_injection",
+            "title": "Testing for Code Injection",
```
Comment (fedetaglia) [inline methodologies/api_testing.json:501]: I can't see this one on the file. but I can see a "Testing for Local File Inclusion" that is not tracked here

Comment (DevAVIJIT7) [reply methodologies/api_testing.json]: Yes, updated the data. Thanks

File: methodologies/api_testing.json (hunk 3 above)
Comment (DevAVIJIT7) [reply methodologies/api_testing.json]: Copy pasting issue. Thanks for pointing.

**Review by fedetaglia — APPROVED** [review]

**Pull request merged**